cirq-google/cirq_google/engine/engine_client.py

```diff
@@ -1090,3 +1106,36 @@ def _date_or_time_to_filter_expr(param_name: str, param: Union[datetime.datetime
         f"type {type(param)}. Supported types: datetime.datetime and"
         f"datetime.date"
     )
+
+
+def _construct_device_config_key(
+    run_name: Optional[str], config_alias: Optional[str]
+) -> Optional[quantum.DeviceConfigKey]:
+    if run_name and not config_alias:
+        raise ValueError("Must specify `config_alias` if `run_name` is non-empty")
+    if run_name is None and config_alias is None:
+        return None
+    return quantum.DeviceConfigKey(run_name=run_name or "", config_alias=config_alias or "")
+
+
+def _construct_processor_selector(
+    project_id: str,
+    processor_id: Optional[str],
+    config_alias: Optional[str] = None,
+    run_name: Optional[str] = None,
+) -> quantum.SchedulingConfig.ProcessorSelector:
+    if processor_id is None:
+        raise ValueError("Processor id must be specified to construct a processor selector")
+    return quantum.SchedulingConfig.ProcessorSelector(
+        processor=_processor_name_from_ids(project_id, processor_id),
+        device_config_key=_construct_device_config_key(run_name, config_alias),
+    )
+
+
+def _fix_processor_ids_kwargs(kwargs):
```

Comment:
This will silently ignore multiple processors if they're specified. I don't expect anyone to use it, but it would be a broken experience if they did. Should we throw an exception in this case?

Reply:
Sounds good to me. I will wait for the answer to this comment, https://github.com/quantumlib/Cirq/pull/6254#discussion_r1302065130 because this wouldn't be necessary if we get rid of the deprecated_parameter decorator.

Reply:
It would be good to also document this, since this would be a breaking change. I think this breaking change is less risky than removing the `processor_ids` positional argument.